### ISSUE TITLE
Unneccessary warning message produced by GroovyDoc task type

### ISSUE THREAD

**Issue by ysb33r** [gradle/gradle#1878]
When running `groovydoc` it always produces a `Trying to override old definition of task fileScanner` message. I cannot remember when this started, but it must have been somewhere in one of the 2.x releases.

It is seemingly harmless, but it detracts from a clean build output.

### Context

This is something people writing Groovy libraries see when generating documentation.

### Steps to Reproduce (for bugs)

```
mkdir example
cd example
gradle init --type groovy-library
./gradlew groovydoc
```

**Comment by ysb33r** [comment]
Original issue mentioned on forum: https://discuss.gradle.org/t/change-in-behaviour-for-groovydoc/18424

**Comment by g1ps** [comment]
...and still happening in 4.9!

**Comment by bjhargrave** [comment]
Still happening in Gradle 6.5.

**Comment by msgilligan** [comment]
Here's a build scan showing the issue: https://scans.gradle.com/s/kbsqbg3tfqecg/console-log#L96

**Comment by epugh** [comment]
Sigh, I get this is as well, and wish there was a way to silence it....

**Comment by ov7a** [comment]
Fixed in 8.14 by
* https://github.com/gradle/gradle/pull/32257
